### PR TITLE
#2305 Implement in-app notifications for non-monetary contributions

### LINKED ISSUE #2305: Notification when new non-monetary contributions are added
### Problem

We don't currently notify group members when a new non-monetary contribution has been added.

### Solution

Post these notifications:

- `{memberName} added non-monetary contribution: {contribution}`
- `{memberName} removed non-monetary contribution: {contribution}`

### PR DESCRIPTION
**Pull request by SebinSong**
closes #2305 

Currently there is two places where users can add/remove/update their non-monetary contributions:
**1.** `/contributions` page.
**2.** `IncomeDetails.vue` modal.

In **1.**, the item is added/removed/updated one by one. Whereas in **2.**, multiple items can be added/removed/updated at a time. In this case, the notification sentence can have compound expressions (not sure if this is a correct english words but anyways) eg. `Teach Javascript and 2 more` or `added Teach Python and removed Teach how to play piano and 1 more` etc.

Clicking on the notification messages navigates to `/contributions` page where they can see the full list of the group's non-monetary contributions.

Hope it looks good.

### CODE REVIEW

File: frontend/model/contracts/group.js [hunk 1]
```diff
@@ -1791,6 +1812,30 @@ sbp('chelonia/defineContract', {
       }).catch(e => {
         console.warn(`removeForeignKeys: ${e.name} error thrown:`, e)
       })
+    },
+    'gi.contracts/group/sendNonMonetaryUpdateNontification': ({
```
Comment (snowteamer) [inline frontend/model/contracts/group.js]: Typo here: -Nontification

Comment (SebinSong) [reply frontend/model/contracts/group.js]: @snowteamer good catch, fixed.

**Comment by SebinSong** [comment]
This PR is ready besides the group-chat heisen-bug

**Comment by taoeffect** [comment]
Thanks @SebinSong. Supposedly @corrideat thinks his PR #2294 (once it's ready) should address the heisenbug.

**Review by taoeffect — APPROVED** [review]
Excellent work @SebinSong! 👏

**Pull request merged**